### ISSUE TITLE
Json pretty printing cannot handle unicode escape character

### ISSUE THREAD

**Issue by dsmt** [serenity-bdd/serenity-core#992]
![](https://d1ro8r1rbfn3jf.cloudfront.net/ms_86337/tRDKAwKdewjhrNCbcMEsLewsamajJm/Json%2Bpretty%2Bprinting%2Bcannot%2Bhandle%2Bunicode%2Bescape%2Bcharacter%2B2017-10-27%2B13-36-25.png?Expires=1509186993&Signature=HyEsQcUEcOO4TL7uTpnaMQ9nr0QGj5Ne7KOYJkC7HDiiUEZIaKrOXx6hLTsBMWvYhyV88nKj65hX5Gq4wXRnm4bCwwaqWX27SUFv~3Uya8jS-cLwFES0tbfxUi8MIgHjB8~9gGAaUZdzpcHJMLtX88eMkD8sp4OEP4S--b6~QTBL6Y1uF2NrWLYK4kXgKbI9-l3eNAvFi6ggdJCNWy-ixkIUZX2xa73frvux8IhnhryRClPnzyEXm8Yq11B~1MHljkmebaEON6HWfNFJQ8iPyMDL9HXrcnRPvyagwWfLoZHFUGwM~Rv2574R~2k0F4ADz97KG3ewnAnc6R34NSr8YA__&Key-Pair-Id=APKAJHEJJBIZWFB73RSA)

![](https://d1ro8r1rbfn3jf.cloudfront.net/ms_86337/FmThxH6I285v3RegaLDPRxPm44emn6/Json%2Bpretty%2Bprinting%2Bcannot%2Bhandle%2Bunicode%2Bescape%2Bcharacter%2B2017-10-27%2B13-37-14.png?Expires=1509187084&Signature=CHJMU3q3arY1l8oROdiLFLzbYofUNiaviLtx0MPeQOr9ohOtzGT2gYiVDTvxo6AFIcfHnZ4WMw3sA0eFWvcbNRdiaWDrdexbmAF8bq1zniWNPMwOqffx3wIYg0~-ilDNPedCChASI7DBuJ1pEDP1huZex9EM3RRPfvemYQzZIx8qFwYG30f2piyZjpAw4~H-WLWsvltXm7UEUal2Xlzw4h1nN9X-2OgMjsyDzm6K3CuJARD4jlaBmznTa1SSxgUNyxvqIqjw9w39NCFJa0eigWo~nR0UBzlzfZryrLtj-nRWulpyAdrgtJj-USJs7Trkt40VMehIzMQ41JI9EzNrFw__&Key-Pair-Id=APKAJHEJJBIZWFB73RSA)

Here is my [sample project](https://github.com/dsmt/serenity-sample-api) for reproducing this defect.

**Comment by wakaleo** [comment]
Fixed with #991

**Comment by dsmt** [comment]
@wakaleo this defect still exists in v1.6.7 and can be reproduced with my sample project

**Comment by wakaleo** [comment]
I'm unclear what behaviour you are expecting here. Could you give an example of a Unicode character that you would input and the output you would expect?

**Comment by dsmt** [comment]
When string in JSON contains `\\u`JSON pretty printing does not work

**Comment by wakaleo** [comment]
Can you give me a concrete example that I can use for the test cases.

**Comment by wakaleo** [comment]
The step headings come from the test code. But the contents of the REST Query dialog come from RestAssured, and represents the actual data that was sent over the wire. At that point, any unicode has already been rendered.
<img width="1062" alt="screen shot 2017-11-01 at 14 33 43" src="https://user-images.githubusercontent.com/55986/32279873-cf8fc15c-bf11-11e7-9dc5-5381b0a5cab9.png">
<img width="959" alt="screen shot 2017-11-01 at 14 33 53" src="https://user-images.githubusercontent.com/55986/32279876-d145e54e-bf11-11e7-8277-720522ee9df0.png">

**Comment by dsmt** [comment]
Sorry for the delay. Let me rephrase, `\\u` character sequence breaks JSON printing in a report. Hope this should help:

```
rest()
                .contentType(ContentType.JSON)
                .body(ImmutableMap.of("string", "\\n"))
                .post("https://jsonplaceholder.typicode.com");
```
![](https://d1ro8r1rbfn3jf.cloudfront.net/ms_86337/gL5JXJKG7nBY3ITwKdlQI9xFq60Wpv/Json%2Bpretty%2Bprinting%2Bcannot%2Bhandle%2Bunicode%2Bescape%2Bcharacter%2B2017-11-01%2B19-24-42.png?Expires=1509639965&Signature=AaEg6d6MWu6DndGruw5SfTNXruYgQndlJY0tVocKiO8luBYTse121aUJyfeRLxyHSlkGB2vj7uTGNTOBkJhuVj6Nk2QE51~VDJE~CO2vPn~pmy3AFpqSwLzZDAwk5pFZao3ME9Iw5CoiR7OTa~5VgG5HEq1DohxajEb9jruD7RtqJzllYGQ5E6yDFqmL3V8iy0AZp0-ueE~QRmy691QWZnk45J6P09XLCKoEr5-50letgM-I7Ioj~57vWcYkkFXmsRdxwxKysdXQOZ89vnHOK9UZCrieVIvxv9DZJG8zEPrKW3OKVYlnsv-6lbrdJLTPplhPGhYxAnR9ny-BJ67Ywg__&Key-Pair-Id=APKAJHEJJBIZWFB73RSA)

```
rest()
                .contentType(ContentType.JSON)
                .body(ImmutableMap.of("string", "\\u"))
                .post("https://jsonplaceholder.typicode.com");
```
![](https://d1ro8r1rbfn3jf.cloudfront.net/ms_86337/rtWxqKWX9DsK7oK2AnRTHYRzdjrEa4/Json%2Bpretty%2Bprinting%2Bcannot%2Bhandle%2Bunicode%2Bescape%2Bcharacter%2B2017-11-01%2B19-27-19.png?Expires=1509640074&Signature=RlpHJL2DCI~2VzGrWk6Vafd7gMQKG5-z~mZk0k~geG4cl~6whvYm~uMI-vdICX3D2rZTJIH9v2OOvEilJF7rAQCtPFBV6P4G4tx5lJv9YDgdLeSrhsXNEAJ~C0D-37rC1gvuPpKFYcHWcCRwj-~xvP2MW1mGtfKqfjkVZacLgJ-gp2ecib1QNyxlN-RY6eGPmJrSwec~nr-hTz9mGd~IqPmAhHk2I-R~NKn4J7K5A8y04iutRIJGZo7SDGdg6ruDZi7RYgKo~5OoL24-l8~jnU4AfmBdLjPVlFTyxj8Pk68VYIYY6beCa5pa-cUHxpW2QfYWnJTPXNU9hql6bUgQow__&Key-Pair-Id=APKAJHEJJBIZWFB73RSA)

**Comment by wakaleo** [comment]
The content is rendered by RestAssured, not by Serenity, so I'm afraid we can't do any better than the reporting in 1.6.8 at this stage. Feel free to have a look at the code and propose a PR if you think you can improve things.

**Comment by dsmt** [comment]
Oh, I see, this is RestAssured issue indeed. Thank you for your time.
